php-checks/src/main/java/org/sonar/php/checks/MultilineBlocksCurlyBracesCheck.java

```diff
@@ -26,104 +26,162 @@
+
+  private static final String MESSAGE_LOOP = "This statement will not be executed in a loop; only the first statement of this %s-statement block will be."
+    + " The rest will execute only once.";
+
+  private static final String MESSAGE_IF = "This statement will not be executed conditionally; only the first statement of this %s-statement block will be."
+    + " The rest will execute unconditionally.";
 
   @Override
   public void visitNode(Tree tree) {
     List<StatementTree> statements = getStatements(tree);
 
     for (int i = 0; i < statements.size() - 1; i++) {
       StatementTree currentStatement = statements.get(i);
-
       if (currentStatement.is(Kind.IF_STATEMENT)) {
-        checkStatement(getLastStatement((IfStatementTree)currentStatement), i, statements);
-
+        IfStatementTree ifStatement = (IfStatementTree) currentStatement;
+        checkStatement(ifStatement.ifToken(), ifStatement.condition(), getLastStatementOfIf(ifStatement), i, statements);
       } else if (currentStatement.is(Kind.FOR_STATEMENT)) {
-        checkStatement(((ForStatementTree) currentStatement).statements().get(0), i, statements);
-
+        ForStatementTree forStatement = (ForStatementTree) currentStatement;
+        checkStatement(forStatement.forToken(), forStatement.closeParenthesisToken(), forStatement.statements().get(0), i, statements);
       } else if (currentStatement.is(Kind.FOREACH_STATEMENT)) {
-        checkStatement(((ForEachStatementTree) currentStatement).statements().get(0), i, statements);
-
+        ForEachStatementTree forStatement = (ForEachStatementTree) currentStatement;
+        checkStatement(forStatement.foreachToken(), forStatement.closeParenthesisToken(), forStatement.statements().get(0), i, statements);
       } else if (currentStatement.is(Kind.WHILE_STATEMENT)) {
-        checkStatement(((WhileStatementTree) currentStatement).statements().get(0), i, statements);
-
+        WhileStatementTree whileStatement = (WhileStatementTree) currentStatement;
+        checkStatement(whileStatement.whileToken(), whileStatement.condition(), whileStatement.statements().get(0), i, statements);
       }
+    }
+  }
 
+  private static StatementTree getLastStatementOfIf(StatementTree statement) {
+    if (!statement.is(Kind.IF_STATEMENT)) {
+      return statement;
     }
+    List<StatementTree> childStatements;
+    IfStatementTree ifStatement = (IfStatementTree) statement;
+    ElseClauseTree elseClause = ifStatement.elseClause();
+    List<ElseifClauseTree> elseifClause = ifStatement.elseifClauses();
+    if (elseClause != null) {
+      childStatements = elseClause.statements();
+    } else if (!elseifClause.isEmpty()) {
+      childStatements = elseifClause.get(elseifClause.size() - 1).statements();
+    } else {
+      childStatements = ifStatement.statements();
+    }
+    return getLastStatementOfIf(childStatements.get(childStatements.size() - 1));
```

Comment:
I don't see the interest of having this method recursive, because I can't see any test cases related to what it could cause.